dashboard/modules/job/job_manager.py

```diff
@@ -517,15 +520,18 @@ async def submit_job(
         # returns immediately and we can catch errors with the actor starting
         # up.
         try:
+            resources = None
+            if _driver_on_current_node:
+                resources = {
+                    self._get_current_node_resource_key(): 0.001,
+                }
             supervisor = self._supervisor_actor_cls.options(
                 lifetime="detached",
                 name=self.JOB_ACTOR_NAME_TEMPLATE.format(job_id=submission_id),
                 num_cpus=0,
                 # Currently we assume JobManager is created by dashboard server
                 # running on headnode, same for job supervisor actors scheduled
-                resources={
-                    self._get_current_node_resource_key(): 0.001,
-                },
+                resources=resources,
```

Comment:
could you change it to affinity scheduling policy here instead of using resources?

Reply:
done.